zokrates_core/src/proof_system/scheme/mod.rs

```diff
@@ -0,0 +1,17 @@
+use proof_system::solidity::SolidityAbi;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use zokrates_field::Field;
+
+pub mod gm17;
+pub mod groth16;
+pub mod pghr13;
+
+pub trait Scheme<T: Field> {
+    type VerificationKey: Serialize + DeserializeOwned;
+    type ProofPoints: Serialize + DeserializeOwned;
+}
+
+pub trait SolidityCompatibleScheme<T: Field>: Scheme<T> {
```

Comment:
Right now it's only Bn128, maybe we need a `SolidityCompatibleField´ trait which only `Bn128Field` implements and then have it as a bound here?

Reply:
Right, I'll do some refactoring on this part as well